ui/admin/app/components/form/role/manage-custom-scopes/index.js

```diff
@@ -5,27 +5,57 @@
 
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 
 export default class FormRoleManageCustomScopesIndexComponent extends Component {
+  // =attributes
+
+  @tracked showRemoveOrgModal = false;
+  @tracked selectedOrg;
+
   // =actions
 
   /**
    * Handles the selection changes for the paginated table.
    * @param {object} selectableRowsStates
    */
   @action
-  selectionChange({ selectableRowsStates }) {
-    const { role } = this.args.model;
+  selectionChange({ selectableRowsStates, selectionKey }) {
+    const { role, projectTotals } = this.args.model;
     selectableRowsStates.forEach((row) => {
       const { isSelected, selectionKey: key } = row;
       const includesId = role.grant_scope_ids.includes(key);
       if (isSelected && !includesId) {
+        // Add the org id if it was selected and it doesn't exist as a grant scope.
         role.grant_scope_ids = [...role.grant_scope_ids, key];
       } else if (!isSelected && includesId) {
+        // Remove org id if it was deselected and it does exist as a grant scope.
         role.grant_scope_ids = role.grant_scope_ids.filter(
           (item) => item !== key,
         );
+        // If the org id was deselected and has selected projects then trigger the modal.
+        const selected = projectTotals[key]?.selected;
+        if (selectionKey === key && selected?.length) {
```

Comment:
I have two questions:
1. I didn't look too closely before but seeing this key logic and looking at the docs I'm now curious, why did we have to iterate over all the `selectableRowsStates` when we get the `selectionKey` for each row? Why was it necessary to iterate over all the rows each time a row is being checked or unchecked? Is it to handle the `all` rows check option?
2. If a user clicks the all checkbox, do we need to handle removing all projects if a user unselects all orgs with it?

Reply:
1. Now that you point it out, there is a more efficient way of doing this by separating the cases that use `all`. I'll experiment with it more.
2. I think that's another good point and i'll check in with design.

Reply:
Update: a new modal has been added for the case in which the user clicks the all checkbox. Thank u for the feedback.

Reply:
@ZedLi tagging for visibility to the comment above